Sources/Order.swift

```diff
@@ -3,12 +3,42 @@ public enum Method: String {
 }
 
 public class Order {
+    public typealias Rules = [(nodeKey: String, rule: EngagementRule)]
     public let path: String
     public let method: Method
 
-    public init(path: String, method: Method = .GET) {
+    /// The rules for the nodes that should be fetched.
+    public let rulesOfEngagement: Rules?
```

Comment:
🎈I don't really get the point of the `rulesOfEngagement`. Is this where you tell the Order if it should parse to a model object?

It would be nice if this could be detected automagically.